### PR TITLE
DEVX-5712 selective stream for archive and broadcast

### PR DESCRIPTION
**Pull request by moficodes**
What is this PR doing?
Adds selective streams to archive and broadcast.

How should this be manually tested?
What are the relevant tickets?
https://jira.vonage.com/browse/DEVX-5712

closing #253 and replacing with this instead

### CODE REVIEW

File: lib/broadcast.js [hunk 1]
```diff
@@ -31,8 +107,8 @@ var Broadcast = function Broadcast(client, json) {
   /**
   * Stops the live streaming broadcast.
   * <p>
-  * Archives automatically stop recording after 120 minutes or when all clients have disconnected
-  * from the session being archived.
+  * broadcasts automatically stop recording after 120 minutes or when all clients have disconnected
+  * from the session being broadcastd.
```
Comment (jeffswartz) [inline lib/broadcast.js:111]:
```suggestion
  * from the session being broadcast.
```

**Review by jeffswartz — APPROVED** [review]

**Comment by moficodes** [comment]
@jeffswartz

File: lib/broadcast.js [hunk 2]
```diff
@@ -7,9 +7,85 @@
...
+*   </ul>
+* @property {Number} maxDuration
+*   Maximum time allowed for the broadcast, in seconds.
+*   After this time, the broadcast will be stopped automatically, if it is still started
+*   Value can be the following:
+*   <ul>
+*     <li>"minimum" -- 60
+*     <li>"maximum" -- 36000
+*     <li>"default" -- 7200
+*   </ul>
```
Comment (jeffswartz) [inline lib/broadcast.js:67]: The value is a number, not a string. Should this be:

```suggestion
You can set the maximum duration to a value from 60 (60 seconds) to
36000 (10 hours). The default maximum duration is 4 hours (14,400 seconds).
```

File: lib/broadcast.js [hunk 3]
```diff
@@ -31,8 +107,8 @@ var Broadcast = function Broadcast(client, json) {
   /**
   * Stops the live streaming broadcast.
   * <p>
-  * Archives automatically stop recording after 120 minutes or when all clients have disconnected
-  * from the session being archived.
+  * broadcasts automatically stop recording after 120 minutes or when all clients have disconnected
```
Comment (jeffswartz) [inline lib/broadcast.js:110]:
```suggestion
  * Broadcasts automatically stop recording after 120 minutes or when all clients have disconnected
```

File: lib/opentok.js [hunk 4]
```diff
@@ -332,6 +345,60 @@ OpenTok = function (apiKey, apiSecret, env) {
    */
   this.listArchives = archiving.listArchives.bind(null, apiConfig);
 
+  /**
+  * Adds a stream to a broadcast that has the streamMode set to manual.
```
Comment (jeffswartz) [inline lib/opentok.js:349]:
```suggestion
  * Adds a stream to an archive that has the streamMode set to manual.
  * You can call the method repeatedly with the same stream ID, to toggle
  * the stream's audio or video in the archive.
```

File: lib/opentok.js [hunk 5]
```diff
@@ -524,13 +605,101 @@ OpenTok = function (apiKey, apiSecret, env) {
...
+      }
       client.startBroadcast(options, function (err, json) {
         if (err) { return callback(new Error('Failed to start broadcast. ' + err)); }
         return callback(null, new Broadcast(client, json));
       });
     }
   };
 
+  /**
+  * Adds a stream to a broadcast that has the streamMode set to manual.
```
Comment (jeffswartz) [inline lib/opentok.js:619]:
```suggestion
```suggestion
  * Adds a stream to a broadcast that has the streamMode set to manual.
  * You can call the method repeatedly with the same stream ID, to toggle
  * the stream's audio or video in the broadcast.
```

**Review by jeffswartz — CHANGES_REQUESTED** [review]

**Review by moficodes — COMMENTED** [review]
/lgtm

**Review by jeffswartz — APPROVED** [review]

**Pull request merged**